### DISCUSSION TITLE
Motor Control.

### DISCUSSION THREAD

**Discussion by KirikauKiwi** [DIY-Flow-Bench/DIY-Flow-Bench#186] [Flow Bench Design]
I wanted to post this as I have not seen much here on how the motors are being controlled , I can see from the code that there is plans for motor control in the DIYFB shield which is great.

- The options I have seen used so far are VFD's on larger motors that are typically 3 phase, as I understand VFD for single phase are not very common, I would like to know more about these in relation to single phase motors.

- Phase angle control along with phased angle solid state relays, manual control via reostats etc.

For my bench design with multiple vacuum cleaner motors it would be great to have sequential switch on as to minimise the start up load which I believe has been thought about looking at the pins.h file, and then the various control methods 0-10V, PWM, RS485.

I see in Europe/UK the phase control from [Kemo](https://www.kemo-electronic.de/en/Light-Sound/Effects/Modules/M028N-Power-control-110-240-V-AC-4000-VA.php) is used commonly, it has an auxiliary device which can then control it via PMW or a voltage.

**Comment by DeeEmm** [comment]
There are a couple of philosophies for pressure regulation. One is to control the motor speed using a VFD, the other is to use a bleed valve. The bleed valve uses a commonly sourced throttle body bypass valve and is stepper motor actuated which can be directly driven by the DIYFB shield.

There are also outputs that allow vac motors to be controlled in banks. Similar to many professional benches. There are three outputs, which can either control three banks of motors OR using an external decoder 2^3 motors. This could be further expanded to 2^4 if the VFD speed ref O/P was reassigned as well. 

Actual control philosophies are still to be finalised but it is likely to be a combination of all of the above. Startup sequencing is of course always a consideration. Reminds me of a job we were commissioning at Rover Cowley where the PLC engineer decided to bring in three massive VFD controlled extraction fans in the paint shop at the same time. Ended up blowing the sub station supply fuse. lol.

At this point in time, allowances have been made for motor control in the shield design, as it needed to be locked in, but the software is not scheduled to be written for those features until after the V2 release.

Regarding speed controllers. They are available for some single phase motors, but they differ from VFDs as they usually control the speed by using a Triac to switch the power off for a portion of each mains cycle. They are essentially glorified dimmers but they work pretty well. The better units have a speed reference input that can be connected to an external reference like the DIYFB shield. (Exactly like the Kemo unit. )  This will interface directly with the Shield using one of the two available on board DACs.

Ultimately the goal is for reference depression regulation to be automatically controlled, which should be relatively trivial. We can regulate the number of active vac motors, along with speed control of one or more motors, plus we can control the bleed valve as well.

It could be that we can operate either a VFD or a bleed valve or that we require some combination of motor and bleed control.

Control is simply a case of bringing in one device / bank at a time and ramping up until we hit the target depression. if we don't hit it then we bring in another device in parallel and ramp up again until we hit the target. If we're not using a VFD then instead we will be regulating the bleed valve instead

**Reply by stefan63** [reply]
To me it feels like blower motor speed is more like a ruff way of controlling pressure and also keepin down the noice level if you have a large vac source, for fine control it feels like a blead valve is faster and more precise. Bu again I'm just braiun storming and have no real hands on experioence...

**Reply by robssmv2** [reply]
Stefan, 

Thats exactly how its done on a regular old school type MSD bench.. But.. Where MSD went wrong is they had no bypass, so it made regulating the depression a real chore by having to force the valves against a deadhead and inevitably it caused flow surging and instability, especially when I started attempting to get to 40inH20. Ended up being nearly impossible to regulate at anything over 32inh2o. Eventually it broke my stainless bowl off the threaded rod. 

I modded my MSD bench with a bypass valve, and, it allowed very repeatable readings that matched the SF1020 within a couple percent every time. Highest I got with the bench was 35inh2o at 300cfm... I could go higher but the motors overheat and shut off. 

hence why I am needing automation. 

Anybody get the closed loop depression control working yet? It wasn't really obvious what part of the code was to be used to it. Was it just the stepper motor control code? 

Would be great if there was a map for the code that showed what did exactly what. I am good, but not that good.

**Reply by DeeEmm** [reply]
@robssmv2 The closed loop control code is not written yet. 

At the moment I'm trying to get the basic functionality and reliability to a point where I can package it up as the official V2 release before moving on to the next round of improvements for V3. This will allow anyone to build the project and get their bench running.

V2 is really just the absolute basics needed to get a functioning and useful system together. V3 is building on that and adding in additional features. There's already a bunch of stuff on the roadmap for V3 onwards including the closed loop control.

You can view the V3 roadmap here - https://github.com/users/DeeEmm/projects/4

If you can think of anything that needs to be added, let me know or feel free to create a feature request or start a discussion in the forums. Once V2 is done and dusted we can prioritise the items on the roadmap in order of what the community wants to see.

> To me it feels like blower motor speed is more like a ruff way of controlling pressure and also keepin down the noice level if you have a large vac source, for fine control it feels like a blead valve is faster and more precise. Bu again I'm just braiun storming and have no real hands on experioence...

Yes, the ability to control depression based on fan speed control will depend entirely on the type and quality of air pump. I say air pump as it would require a positive displacement pump like a rootes blower or hydrovane. This way the air can be accurately metered by modulating the air pump speed. As this kind of setup is not going to be what most people build, instead going the pathway of regular vacuum motors, it is most likely that for a majority of cases, the bleed valve is the way to go.

Control will likely be something like....

- Bleed valve is closed
- Motor(s) ramp up until target depression is reached + an overshoot factor
- Bleed valve is then opened to regulate depression back down to target

Then when running. If depression drops, for example when opening up the valve...

- Bleed valve will automatically modulate to closed position trying to reach target depression.
- When bleed valve close position is detected, motors ramp up to maintain target + overshoot factor
- Bleed valve then opens to regulate depression back down to target

The actual bleed valve modulation will use PID control.

This is a basic cyclical closed loop control philosophy. You can also replace motor VFD control with vac motors wired in banks. This will be a little different in that the bank of motors cannot ramp up, so the bleed valve starts opened and closes down to set the target.

Then in addition to this we will also need a manual based setup in which the vac motors are manually controlled (either in banks of by VFD).

In all cases the bleed valve automatically regulates to set the target.

@robssmv2. Achieving 40" + is easily possible, but I would not use vac motors. I would use a large blower driven by an electric motor controlled by VFD. Early on in development when I was trying out various vac sources, I did some tests using a small blower driven by an electric motor and the results were impressive . (my main business is making supercharger kits for vintage VW's). I easily exceeded the limits of my test equipment. 

Generic industrial Rootes blowers list +/-400"/WG in the specs, which is about 1 bar. When you think of a blower making 1 bar of boost or more over the top of what the engine is consuming you can see that it is easily capable of moving enough air to test at whatever depression you want. 40" should be easily achievable. 

A blower powered bench is something that I intend to revisit down the track. It is also heaps quieter than a vac style bench too.

![IMG_4739](https://github.com/user-attachments/assets/5a6bff6b-9ae4-46d0-b8f2-4110c735c090)
